metadata.rb

```diff
@@ -21,6 +21,7 @@
 depends          'apt' # We recommend '>= 2.1.0'. See CHANGELOG.md for details
 depends          'chef_handler', '~> 1.1.0'
 depends          'yum'
+depends          'windows'
```

Comment:
I wanted to use the default providers at first but ran into a lot of issues with the `windows_package` from core :
- the resource is broken in chef 12.2 and 12.3 (fixed in 12.4), see https://github.com/chef/chef/issues/3316
- using the resource with `action :remove` to uninstall the agent makes the recipe fail if the agent is not installed, instead of simply being up-to-date 
- the resource can upgrade the agent but sometimes breaks it in the upgrade (for instance the upgrade from 5.3.0 to 5.3.1 breaks the agent)
- the original msi needs to be provided to uninstall the agent (whereas the `windows` cookbook handles that nicely by using registry keys to find the uninstaller)

I haven't had any of these issues with the `windows` cookbook.
It seems that the `windows` cookbook is gradually getting merged in chef core, until then I think we should use the cookbook.